Apps/Common/src/AccessManagement/Administration/KeycloakUserAdminDelegate.cs

```diff
@@ -0,0 +1,148 @@
+        /// </summary>
+        public const string KEYCLOAKADMIN = "KeycloakAdmin";
+
+        /// <summary>
+        /// Configuration Key for the Delete User Url.
+        /// </summary>
+        public const string DELETEUSERURL = "DeleteUserUrl";
+
+        /// <summary>
+        /// Configuration Key for the Get User Url.
+        /// </summary>
+        public const string GETUSERURL = "GetUserUrl";
+
+        /// <summary>
+        /// The injected logger delegate.
+        /// </summary>
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// The injected HttpClientService delegate.
+        /// </summary>
+        private readonly IHttpClientService httpClientService;
+
+        /// <summary>
+        /// The injected configuration delegate.
+        /// </summary>
+        private readonly IConfiguration configuration;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KeycloakUserAdminDelegate"/> class.
+        /// </summary>
+        /// <param name="logger">The injected logger provider.</param>
+        /// <param name="httpClientService">injected HTTP client service.</param>
+        /// <param name="configuration">Injected configuration.</param>
+        public KeycloakUserAdminDelegate(
+            ILogger<KeycloakUserAdminDelegate> logger,
+            IHttpClientService httpClientService,
+            IConfiguration configuration)
+        {
+            this.logger = logger;
+            this.httpClientService = httpClientService!;
+            this.configuration = configuration;
+        }
+
+        /// <inheritdoc/>
+        public UserRepresentation GetUser(Guid userId, JWTModel jwtModel)
+        {
+            this.logger.LogInformation($"Keycloak GetUser : {userId.ToString()}");
+
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public int DeleteUser(Guid userId, JWTModel jwtModel)
+        {
+            this.logger.LogInformation($"Keycloak DeleteUser : {userId.ToString()}");
+
+            Task<int> task = Task.Run(async() => await this.DeleteUserAsync(userId.ToString(), jwtModel).ConfigureAwait(true));
+            task.Wait();
+            if (task.Exception != null)
+            {
+                throw new Exception("KeycloakUserAdminDelegate.DeleteUser() exception", task.Exception);
+            }
+            return task.Result;
+
+        }
+
+        /// <summary>
+        /// Deletes the User account from Keycloak
+        /// </summary>
+        /// <param name="userId">The user id to delete.</param>
+        /// <param name="jwtModel">To get at the base64 access token</param>
+        private async Task<int> DeleteUserAsync(string userId, JWTModel jwtModel)
+        {
+
+            Uri baseUri = new Uri(this.configuration.GetSection(KEYCLOAKADMIN).GetValue<string>(DELETEUSERURL));
+
+            using HttpClient client = this.CreateHttpClient(baseUri, jwtModel.AccessToken!);
+            {
```

Comment:
Why the brackets?

Reply:
was legacy using conversion not needed... wil remove